JoinColony/colonyDapp
# Connect transfer nft

## Description

This PR wires up the NFT Transfer function.

ERC721 abi will be added to colonyJS. It's been included here temporarily and will be removed once it can be accessed from colonyJS. 

**To test:**

1. Follow [the relevant setup instructions](https://github.com/JoinColony/colonyDapp/pull/3765) to get the foreign bridge running. Ensure you pull the latest changes from `maint/cross-chain-safe`.
2. Ensure LOCAL_SAFE_ADDRESS= and LOCAL_ERC721_ADDRESS= are added to your `.env`. Restart webpack. 
3. Go to  add existing safe.
4. Add a safe with nfts in it (e.g. [0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27](https://gnosis-safe.io/app/rin:0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27). Not important for testing since NFT is deployed locally.)
5. Go to safe control and select  "transfer NFT".  Select an NFT. Select a recipient. 
6.  Confirm the transaction. 
7. In the terminal window in which you're running the foreign bridge, you should see a confirmation that the NFT has been transferred out of the safe. In the console in the browser, you should see confirmation that the bridging transaction was successful.


**Changes** 🏗

* Adapting saga for all transaction types and adding the NFT Transfer function. 
* Adding helpers to this end. 

Supported by [this commit](https://github.com/JoinColony/colonyNetwork/commit/66ab664eb400bb03010ed394cdcd4985c147f23d) in ColonyNetwork (adding and deploying ERC721 mock).

Resolves #3722

## Setup screenshot

You will see this if your bridging env is set up correctly. Ensure your addresses are correct in your `.env`:

![success](https://user-images.githubusercontent.com/64402732/191469844-8d0aeb72-dbf2-4573-92a5-ab67f7af801e.png)

And this if the NFT transfer goes through:

![works](https://user-images.githubusercontent.com/64402732/191470727-039c650c-293c-4735-bd77-6d00e2132f40.png)
